global_planner/src/orientation_filter.cpp

```diff
@@ -97,6 +97,29 @@ void OrientationFilter::processPath(const geometry_msgs::PoseStamped& start,
                 }
             }
             break;
+        case SIDEWARD:
+            for (int i = 0; i < n - 1; i++) {
+                setAngleBasedOnPositionDerivative(path, i);
+            }
+            if (n > 2) {
+                const int num_skips = n >= 5 ? 2 : 1;
+                const double start_path_theta = tf2::getYaw(path[num_skips].pose.orientation);
+                const double goal_path_theta = tf2::getYaw(path[n - 1 - num_skips].pose.orientation);
+                const double start_orientation = tf2::getYaw(start.pose.orientation);
+                const double rotation_to_path = angles::shortest_angular_distance(start_orientation, start_path_theta);
+                const double rotation_to_goal = angles::shortest_angular_distance(start_orientation, goal_path_theta);
+
+                bool prefer_leftward = 
+                    std::fabs(angles::normalize_angle(rotation_to_path - M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal - M_PI_2)) < 
+                    std::fabs(angles::normalize_angle(rotation_to_path + M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal + M_PI_2));
```

Comment:
F: I think the second time, the sign of the half rotation needs to be inverted, no? I.e.
```suggestion
                    std::fabs(angles::normalize_angle(rotation_to_path - M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal + M_PI_2)) < 
                    std::fabs(angles::normalize_angle(rotation_to_path + M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal - M_PI_2));
```

Reply:
It should be inverted, that is correct, thanks. The robot will need to undo the rotation.

![image](https://github.com/rapyuta-robotics/ros-navigation/assets/6097267/0d87e4a5-0c39-4137-bc7c-48b0aded7117)

Reply:
nice picture. Is that the plot of 
```
std::fabs(angles::normalize_angle(rotation_to_path - M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal + M_PI_2)) < 
std::fabs(angles::normalize_angle(rotation_to_path + M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal - M_PI_2));
```
I suppose? Is there some intuitive way to see that this is indeed the correct result?

Reply:
yes. that is the plot of the function:
```
std::fabs(angles::normalize_angle(rotation_to_path - M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal + M_PI_2)) < 
std::fabs(angles::normalize_angle(rotation_to_path + M_PI_2)) + std::fabs(angles::normalize_angle(rotation_to_goal - M_PI_2));
```

hm, I think the best way to understand is by visualizing the start/goal in circle quadrant:

![image](https://github.com/rapyuta-robotics/ros-navigation/assets/6097267/e1805350-f0a0-4d80-92a5-c5ae4a614171)

The cases we want leftward/rightward (figures on the left), we can see that the solution is the correct one. 
The other cases (when we switch orientation by a lot) it doesn't matter much which one we should choose since the "cost" is basically the same (for both leftward or rightward), and that is why we have this "frog" graph.
My concern is with the switching part, which can lead to some oscillation, maybe